pkgs/by-name/im/immich/package.nix

```diff
@@ -181,7 +180,20 @@ buildNpmPackage' {
   # Required because vips tries to write to the cache dir
   makeCacheWritable = true;
 
+  preConfigure = ''
+    # we manually build sharp from source later on
+    # FIXME figure out why otherwise it fails with
+    #     error: 'NewOrCopy' is not a member of 'Napi::Buffer<char>'
+    export SHARP_IGNORE_GLOBAL_LIBVIPS=1
+  '';
+
   preBuild = ''
+    unset SHARP_IGNORE_GLOBAL_LIBVIPS
+    export SHARP_FORCE_GLOBAL_LIBVIPS=1
+
+    rm -r node_modules/node-addon-api
+    rm -r node_modules/**/node-addon-api
```

Comment:
@vale981 That's not an issue since I remove all old versions of node-addon-api here.

Reply:
with renaming preConfigure to prePatch, I get it to build but then it fails with
```
       > Finished npmBuildHook
       > Running phase: glibPreInstallPhase
       > Running phase: glibPreInstallPhase
       > Running phase: installPhase
       > npm error code ENOTCACHED
       > npm error request to https://registry.npmjs.org/node-addon-api/-/node-addon-api-5.1.0.tgz failed: cache mode is 'only-if-cached' but no cached response is available.
       > npm error A complete log of this run can be found in: /build/.npm/_logs/2025-01-18T01_34_28_575Z-debug-0.log
       For full logs, run 'nix log /nix/store/l1jg60arlbd2y1v7bi10hj7lcqfqh8gq-immich-1.124.2.drv'.
```